src/components/Effects.tsx

```diff
@@ -66,6 +72,9 @@ export const Effects: FC = () => {
           }
 
           if (shouldLogout) {
+            // setting the state synchronously may cause problems (not really sure), so we add a small delay
+            await new Promise((resolve) => setTimeout(resolve, 50))
```

Comment:
A bit hacky. Add a `FIXME` for clarification & notice for future maintainers.

```suggestion
            // FIXME: setting the state synchronously may cause problems (not really sure), so we add a small delay for a temporary hack
            await new Promise((resolve) => setTimeout(resolve, 50))
```